mava/types.py

```diff
@@ -21,17 +21,58 @@
 from optax._src.base import OptState
 from typing_extensions import NamedTuple, TypeAlias
 
-from mava.wrappers.jumanji import LogEnvState
+if TYPE_CHECKING:  # https://github.com/python/mypy/issues/6239
+    from dataclasses import dataclass
+else:
+    from flax.struct import dataclass
+
 
 Action: TypeAlias = chex.Array
 Value: TypeAlias = chex.Array
 Done: TypeAlias = chex.Array
 HiddenState: TypeAlias = chex.Array
-
-# Can't know the exact type of State or Timestep.
+# Can't know the exact type of State.
 State: TypeAlias = Any
-Observation: TypeAlias = Any
+
+
+class Observation(NamedTuple):
```

Comment:
Could we take the docs from `ObservationGlobalState` and put them here and then only put the `global_state` doc in the observation global state.
```suggestion
class Observation(NamedTuple):
"""The observation that the agents see. Used for all decentralized systems.

    agents_view: the array that is passed to the networks during acting and learning.
    action_mask: boolean array specifying, for each agent, which action is legal.
    step_count: the number of steps elapsed since the beginning of the episode.
```